apps/src/code-studio/pd/workshop_dashboard/components/facilitator_list_form_part.jsx

```diff
@@ -60,14 +60,18 @@ export default class FacilitatorListFormPart extends React.Component {
       this.props.facilitators.push({id: -1});
     }
 
+    let filteredAvailableFacilitators = null;
+    // Remove already-selected facilitators from available list.
+    if (!this.props.readOnly) {
+      filteredAvailableFacilitators = this.props.availableFacilitators.filter(availableFacilitator => {
+        return !this.props.facilitators.find(f => f.id === availableFacilitator.id);
+      });
+    }
```

Comment:
Moving this out of the `map` on line 71 is just an optimization, not related to the bugs, but it seems like a good idea. There's really no need to recompute this available list for each row.

Reply:
:+1: Good extraction.  One syntax nit - could express the outer arrow function here with an expression too.
```jsx
filteredAvailableFacilitators = this.props.availableFacilitators.filter(availableFacilitator =>
  !this.props.facilitators.find(f => f.id === availableFacilitator.id);
);
```

Reply:
ah right, good call 👍  Will do